sockeye/train.py

```diff
@@ -741,22 +784,27 @@ def main():
 
         shared_vocab = use_shared_vocab(args)
 
-        train_iter, eval_iter, config_data, vocab_source, vocab_target = create_data_iters_and_vocab(
+        train_iter, eval_iter, config_data, vocab_source, vocab_target, source_factor_vocabs = create_data_iters_and_vocabs(
             args=args,
             shared_vocab=shared_vocab,
             resume_training=resume_training,
             output_folder=output_folder)
 
+        # Dump the vocabularies if we're just starting up
         if not resume_training:
             vocab.vocab_to_json(vocab_source, os.path.join(output_folder, C.VOCAB_SRC_NAME) + C.JSON_SUFFIX)
             vocab.vocab_to_json(vocab_target, os.path.join(output_folder, C.VOCAB_TRG_NAME) + C.JSON_SUFFIX)
+            for i, v in enumerate(source_factor_vocabs):
+                vocab.vocab_to_json(v, os.path.join(output_folder, C.VOCAB_SRC_NAME) + "." + str(i) + C.JSON_SUFFIX)
 
         vocab_source_size = len(vocab_source)
         vocab_target_size = len(vocab_target)
-        logger.info("Vocabulary sizes: source=%d target=%d", vocab_source_size, vocab_target_size)
+        source_factor_vocab_sizes = [len(v) for v in source_factor_vocabs]
+        logger.info("Vocabulary sizes: source (%s) target (%s)",
```

Comment:
this seems hard to understand when read in a logfile. Can we make this more explicit and log something like:
"Vocabulary sizes: source=%d source_factors=%d target=%d"?

Reply:
changed a bit